### PR TITLE
add 3 new lumi sensors

### PR DESCRIPTION
**Pull request by zhouF96**
P_20200602_135545
P_20200602_135600

### CODE REVIEW

**Comment by zhouF96** [comment]
The support for WSDCGQ12LM is still experimental as i can't get the configure key for it. but it can work with a wrong configure key ? don't know why. and attached the sniffed packet here
[z2m wsdcgq12lm.zip](https://github.com/Koenkk/zigbee-herdsman-converters/files/4715280/z2m.wsdcgq12lm.zip)

**Comment by Koenkk** [comment]
What is wrong with the WSDCGQ12LM? The configureKey only indicates if the configure() method has been changed (so that it needs to be re-executed): https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/devices.js#L5

File: devices.js [hunk 1]
```diff
@@ -814,6 +830,15 @@ const devices = [
         fromZigbee: [fz.xiaomi_battery_3v, fz.SJCGQ11LM_water_leak_iaszone],
         toZigbee: [],
     },
+    {
+        zigbeeModel: ['lumi.flood.agl02'],
+        model: 'SJCGQ12LM',
+        vendor: 'Xiaomi',
+        description: 'Aqara water leak sensor',
+        supports: 'water leak true/false',
+        fromZigbee: [fz.xiaomi_battery_3v, fz.SJCGQ12LM_water_leak_iaszone],
```
Comment (Koenkk) [inline devices.js:839]: Please try with `fz.ias_water_leak_alarm_1` instead of `fz. SJCGQ12LM_water_leak_iaszone`

File: devices.js [hunk 2]
```diff
@@ -942,6 +967,15 @@ const devices = [
         fromZigbee: [fz.xiaomi_battery_3v, fz.DJT11LM_vibration],
         toZigbee: [tz.DJT11LM_vibration_sensitivity],
     },
+    {
+        zigbeeModel: ['lumi.vibration.agl01'],
+        model: 'DJT12LM',
+        vendor: 'Xiaomi',
+        description: 'Aqara vibration sensor',
+        supports: 'drop, tilt and touch',
+        fromZigbee: [fz.genOnOff_cmdOn],
```
Comment (Koenkk) [inline devices.js:976]: returning `click: on` makes no sense for this device, does `fz.DJT11LM_vibration` work?

Comment (zhouF96) [reply devices.js:976]: by default  it returns click: on once the device reveives vibration three times in a  short time only and  fz.DJT11LM_vibration will not work(but can be configured by using aqara's new gateway and their Aqara home app ). And aqara said they will not opensource shortly and the samples i got were only engineer samples.  Currently I don't have enough time to figure out

File: devices.js (hunk 1 above)
Comment (zhouF96) [reply devices.js:839]: tried fz.ias_water_leak_alarm_1 instead of fz. SJCGQ12LM_water_leak_iaszone  works perfectly.

File: devices.js (hunk 2 above)
Comment (Koenkk) [reply devices.js:976]: In that case return `{"action": "vibration"}`, that makes much more sense then click: on.

**Comment by Koenkk** [comment]
> What is wrong with the WSDCGQ12LM? The configureKey only indicates if the configure() method has been changed (so that it needs to be re-executed): https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/devices.js#L5

@zhouF96 please also check this

**Comment by zhouF96** [comment]
confirmed that if do not use 
`
        meta: {configureKey: 1},
        configure: async (device, coordinatorEndpoint) => {
            const endpoint = device.getEndpoint(1);
            const binds = ['msTemperatureMeasurement', 'msRelativeHumidity', 'msPressureMeasurement'];
            await bind(endpoint, coordinatorEndpoint, binds);
        },
`
the device will not return any tempera,humidity  nor pressure message.
and with configure , it can return those sensor values correctly.

log with configure key here (please ignore the error not connect to a mqtt broker)

[log.txt](https://github.com/Koenkk/zigbee-herdsman-converters/files/4730960/log.txt)


also changed the DJT12LM and will return action: 'vibration' if received vibration three times in a short time. 
BUT it still need to add support for this devce to change the default behavior to let it behave like a common vibration sensor.

File: devices.js [hunk 3]
```diff
@@ -760,6 +760,22 @@ const devices = [
         ],
         toZigbee: [],
     },
+    {
+        zigbeeModel: ['lumi.sensor_ht.agl02'],
+        model: 'WSDCGQ12LM',
+        vendor: 'Xiaomi',
+        description: 'Aqara temperature, humidity and pressure sensor',
+        supports: 'temperature, humidity and pressure',
+        fromZigbee: [fz.xiaomi_battery_3v, fz.xiaomi_temperature, fz.humidity, fz.WSDCGQ11LM_pressure,
```
Comment (Koenkk) [inline devices.js:769]: Can you remove/replace the following converters and check if pressure and temperature still work ok?
- `xiaomi_temperature` -> `temperature`
- `WSDCGQ11LM_pressure` -> `pressure`.
- Remove `WSDCGQ01LM_WSDCGQ11LM_interval` (not sure if this is used?)

File: devices.js [hunk 4]
```diff
@@ -942,6 +967,15 @@ const devices = [
...
         toZigbee: [tz.DJT11LM_vibration_sensitivity],
     },
+    {
+        zigbeeModel: ['lumi.vibration.agl01'],
+        model: 'DJT12LM',
+        vendor: 'Xiaomi',
+        description: 'Aqara vibration sensor',
+        supports: 'drop, tilt and touch',
+        fromZigbee: [fz.DJT12LM_vibration],
+        toZigbee: [tz.DJT11LM_vibration_sensitivity],
```
Comment (Koenkk) [inline devices.js:977]: Does setting the sensitivity work?

File: devices.js (hunk 3 above)
Comment (zhouF96) [reply devices.js]: removed WSDCGQ01LM_WSDCGQ11LM_interval. replaced xiaomi_temperature -> temperature and WSDCGQ11LM_pressure -> pressure  can work properly

File: devices.js (hunk 4 above)
Comment (zhouF96) [reply devices.js]: setting the sensitivity can **not**  work properly , So removed and remain for someone else to commit for this.

**Comment by Koenkk** [comment]
Thanks!

**Comment by dmulcahey** [comment]
@zhouF96 where did you find these? I’m looking to get several of the T1 variants to add support to ZHA

**Comment by zhouF96** [comment]
> @zhouF96 where did you find these? I’m looking to get several of the T1 variants to add support to ZHA

Aqara send us some engineer samples for testing with [scratch3](https://scratch.codelab.club/projects/editor/) and [codelab adapter](https://www-old.codelab.club/blog/codelab-download/) to use it to code your space. They have no plan to opensource it currently and those T1 variants are not sale online. If you need , I can send you some , but Postage needed.

**Comment by dmulcahey** [comment]
Can you put me in touch with Aqara? I don’t mind working directly with them... if not, Let’s find a way to discuss this off this thread. I’m on discord with the same handle.

**Comment by zhouF96** [comment]
> Can you put me in touch with Aqara? I don’t mind working directly with them... if not, Let’s find a way to discuss this off this thread. I’m on discord with the same handle.

I've contacted with Aqara and posted your contact info and github in wechat, but It is midnight in China, They may reply to you tomorrow? Once have any info , I'll send your emails.

**Comment by dmulcahey** [comment]
Thanks!!

**Comment by Wh1terat** [comment]
> also changed the DJT12LM and will return action: 'vibration' if received vibration three times in a short time. BUT it still need to add support for this devce to change the default behavior to let it behave like a common vibration sensor.

@zhouF96 
Did you ever find solution for additional attributes? (tilt, etc)

DJT12LM seems quite useless vs DJT11LM

**Pull request merged**